src/cmds/restic/cmd_backup.go

```diff
@@ -293,6 +303,10 @@ func readLinesFromFile(filename string) ([]string, error) {
 }
 
 func runBackup(opts BackupOptions, gopts GlobalOptions, args []string) error {
+	if opts.FilesFrom == "-" && gopts.password == "" && gopts.PasswordFile == "" && os.Getenv("RESTIC_PASSWORD") == "" {
```

Comment:
It's enough to test for `gopts.password == ""`, that field is filled with the contents of the environment variable on startup (here: https://github.com/restic/restic/blob/master/src/cmds/restic/global.go#L46-L49), so you can remove the test for `os.Getenv(...) == ''`.